### ISSUE TITLE
please add tabindex=-1 to wrapping a-tags of InputfieldButton

### ISSUE THREAD

**Issue by BernhardBaumrock** [processwire/processwire-issues#1220]
Take this button:

```php
$b = $modules->get('InputfieldButton');
$b->icon = 'check';
$b->value = 'foo';
$b->href = '/bar';
$b->attr('uk-tooltip', 'My great button tooltip');
echo $b->render();
```

This will show a nice uikit tooltip when hovering the button. Thanks to uikit's good quality this does also work using the keyboard, but ProcessWire breaks accessibility a little because the rendered button has a wrapping anchor tag that adds an additional required TAB before actually focusing the button that shows the tooltip.

Solution: Simply adding `tabindex="-1"` to the wrapping anchor solves the problem.

![image](https://user-images.githubusercontent.com/8488586/88943206-7c8b1c00-d28b-11ea-815f-ed420002f360.png)

**Comment by gadgetto** [comment]
I think it would be much better to solve this by not wrapping a button element with an a tag. This is simply not valid HTML5! Some time ago, I created an issue regarding this problem: https://github.com/processwire/processwire-issues/issues/873

**Comment by BernhardBaumrock** [comment]
Thx @gadgetto I know about the issue you created. You might be right that this markup is not valid, though I'd be happy to get this fix as an easy improvement to an existing (and working) solution.

Changing the markup as you suggest would have a lot of unwanted side effects I fear. Adding tabindex -1 should on the other hand be quite simple (I only tested it with my boring buttons in the screenshot though).

Also I have to say that I'm not sure if removing the anchor would really be a good idea... does middle click for opening in a new tab even work? But that's offtopic and I don't want to start a discussion here...

**Comment by matjazpotocnik** [comment]
While we could hook to InputfieldButton::render() and change/replace the markup, I would also like to see a proper solution. But, as Bernhard pointed out, we don't know if the change in the markup would cause trouble, so adding tabindex=-1 is probably the only solution at the moment.

@BernhardBaumrock, are you sure that adding tabindex=-1 to `<a>` solves a problem? Not in my case, I had to add tabindex=-1 to the `<button>`.

**Comment by ryancramerdesign** [comment]
@BernhardBaumrock @matjazpotocnik @gadgetto 

I have added "tabindex=-1" to the `<a>` element which seems to work here. I also tried it on the button element, but couldn't get that to work quite right.

I've also added a `linkInner` option that lets you specify you want the `<a>` inside the button rather than outside. I found it wasn't working well with the admin themes, but is now there as an option for people that might want to use this Inputfield elsewhere. Perhaps we can eventually transition to it on the admin side when it fits alongside other admin updates.
